src/intersections/triangles.jl

```diff
@@ -143,19 +143,17 @@ and apply function `f` to it.
 """
 function intersecttype(f::Function, r::Ray{3,T}, t::Triangle{3,T}) where {T}
   vs = vertices(t)
-  o = r.p
-  d = r.v
 
   e₁ = vs[3] - vs[1]
   e₂ = vs[2] - vs[1]
-  p = d × e₂
+  p = direction(r) × e₂
```

Comment:
You don't need to change the code here, you can just redefine the variables `o` and `d` using the public API now.